packages/terra-data-grid/src/DataGrid.jsx

```diff
@@ -327,8 +347,18 @@ const DataGrid = forwardRef((props, ref) => {
     if (!isSection(toCell.row)) {
     // Obtain coordinate rectangles for grid container, column header, and new cell selection
       const gridContainerRect = tableContainerRef.current.getBoundingClientRect();
-      const columnHeaderRect = grid.current.rows[0].cells[toCell.col].getBoundingClientRect();
+      const columnHeaderRect = grid.current.rows[0].getBoundingClientRect();
+
+      // Set new cell column index value to be within Column Header and Column Header actions bounding indexes 
+      if(toCell.row === 0 && hasVisibleColumnHeaders && toCell.col > displayedColumns.length) {
```

Comment:
Why would toCell.col be greater than the number of column headers?  There seems to be a logical error in the design somewhere.